crates/codegen/parser/runtime/src/mod_for_destination.rs

```diff
@@ -1,15 +1,69 @@
 #[macro_use]
 pub mod parser_support;
 
-pub mod cst;
-pub mod cursor;
 pub mod kinds;
 pub mod language;
 pub(crate) mod lexer;
 pub mod parse_error;
 pub mod parse_output;
-pub mod query;
-pub mod text_index;
+pub mod queries;
 
 #[cfg(feature = "slang_napi_interfaces")]
 pub mod napi_interface;
+
+mod metaslang_cst {
```

Comment:
This `mod metaslang_cst` has the same name as the newly added crate. Is this required? Not sure I understand why `KindTypes` is wrapped here in a private module.

Since this type is exposed in the public API through other types, I suggest adding it to `crate::kinds` module, along other kinds.

Reply:
This is the same name but a different path. It is the parameter for the import of the generic module. I am using an ML Functor model, where the import is a type-level function.

To contextualize the names, which I think is a good idea, I put them in this local module.

I have a strong instinct that it shouldn't be exposed in the kinds module, because these are not kind definitions, Functor params.